08-compiler-features.tex

```diff
@@ -142,81 +142,347 @@ \section{Dead Code Elimination}
+	\item The output is HTML-escaped so that \ic{\&}, \ic{<} and \ic{>} become \ic{\&amp;}, \ic{\&lt;} and \ic{\&gt;} respectively.
+	\item Otherwise any documentation output is preserved which means longer documentation might include new-line and tab-characters as it does in the source files.
+	\item When run in completion mode, the compiler does not display errors but instead tries to ignore them or recover from them.  If a critical error occurs while getting completion, the Haxe Compiler prints the error message instead of the completion output. Any non-XML output can be treated as a critical error message.
+\end{itemize}
+
+\subsection{Field access completion}
+\label{cr-completion-field-access}
+
+Field completion is triggered after a dot \ic{.} character to list the fields that are available on the given type. The compiler parses and types everything up to the point of completion and then outputs the relevant information to stderr:
 
 \begin{lstlisting}
-class Test {
-    public static function main() {
-        trace("Hello".|
-    }
+class Main {
+  public static function main() {
+    trace("Hello".|
+  }
+}
+\end{lstlisting}
+
+If this file is saved to Main.hx the completion can be invoked using the command \ic{haxe --display Main.hx@0}. The output looks similar to this (we omit several fields for brevity and improve the formatting for readability):
+
+\begin{lstlisting}
+<list>
+<i n="length">
+  <t>Int</t>
+  <d>
+    The number of characters in `this` String.
+  </d>
+</i>
+<i n="charAt">
+  <t>index : Int -&gt; String</t>
+  <d>
+    Returns the character at position `index` of `this` String.
+    If `index` is negative or exceeds `this.length`, the empty String
+    "" is returned.
+  </d>
+</i>
+<i n="charCodeAt">
+  <t>index : Int -&gt; Null&lt;Int&gt;</t>
+  <d>
+    Returns the character code at position `index` of `this` String.
+    If `index` is negative or exceeds `this.length`, null is returned.
+    To obtain the character code of a single character, "x".code can
+    be used instead to inline the character code at compile time.
+    Note that this only works on String literals of length 1.
+  </d>
+</i>
+</list>
+\end{lstlisting}
+
+The XML structure follows:
+
+\begin{itemize}
+	\item The document node \ic{list} encloses several nodes \ic{i}, each of which representing a field.
+	\item The \ic{n} attribute contains the name of the field.
+	\item The \ic{t} node contains the type of the field.
+	\item the \ic{d} node contains the documentation of the field.
+\end{itemize}
+
+\since{3.2.0}
+
+When compiling with \ic{-D display-details} each field additionally has a \ic{k} attribute which can either be \ic{var} or \ic{method}. This allows distinguishing method fields from variable fields that have a function type.
```

Comment:
comma after "display details" consider: "each field has an additional attribute" or "In addition, each field"